### PR TITLE
Binja database support

### PR DESCRIPTION
**Pull request by xusheng6**
This PR adds support for loading and analyzing a binja database: https://github.com/mandiant/capa/issues/2496

There are two things I wish to mention:

1. This PR is rebased on https://github.com/mandiant/capa/pull/2500, rather then the latest master. I now realize this is not the best thing to do, but I cannot simply drop the commits from #2500 because there are conflicts. So maybe first have a look at that PR, and if it gets approved, I can rebase this again (if needed)
2. I did not add a unit test for this. I know I can save one of the file we use for unit tests to a binja database and then load it. Do you have a representative binary to use for it among those files? Also, should I add a new test or should I just add it to `test_binja_features.py`?

### Checklist

- [ ] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

**Review by williballenthin — APPROVED** [review]

**Comment by williballenthin** [comment]
> Do you have a representative binary to use for it among those files? Also, should I add a new test or should I just add it to test_binja_features.py?

I think you should take a trivial file, like PMA 01-01.dll, and use that. The test only needs to show that the db file can be analyzed, not that all the features are extracted consistently (which we can assume will be the case, I think).

File: capa/features/extractors/binja/find_binja_api.py [hunk 1]
```diff
@@ -105,13 +105,13 @@ def find_binaryninja() -> Optional[Path]:
             logger.debug("detected OS: linux")
         elif sys.platform == "darwin":
             logger.warning("unsupported platform to find Binary Ninja: %s", sys.platform)
-            return False
+            return None
```
Comment (williballenthin) [inline capa/features/extractors/binja/find_binja_api.py:108]: good catch

**Review by mr-tz — APPROVED** [review]

**Comment by mr-tz** [comment]
Pending tests

**Comment by xusheng6** [comment]
> Pending tests

I added a test for for it, please have a look!

**Comment by xusheng6** [comment]
Oh well apparently I only added the database file and the feature list, but forgot to make it actually run during the test. I will push a fix shortly

**Review by mr-tz — APPROVED** [review]
Great, thank you!

**Review by williballenthin — APPROVED** [review]

**Comment by xusheng6** [comment]
Just so that you know this branch is failing the (newly added) unit test (sorry I wrote the code without running the unit test, I thought it is way to easy to bother with testing at all)

All new binja db tests are failing because it cannot properly get a binary view, I am yet to see what is happening. It does work perfectly fine when I analyze a standalone file outside of the unit test infra

**Comment by xusheng6** [comment]
@williballenthin oh well it is just a simple typo in the file path. I have fixed it and this should be ready for merge

**Comment by xusheng6** [comment]
@mr-tz thx for merging this! However, since I also introduced a submodule change on the test files, now the capa repo points to this commit https://github.com/mandiant/capa-testfiles/tree/b40457c616604ee65f44e1ae8597cfeec7da0f70 which is not on top of the master branch of the capa-testfiles repo. I believe you should cherry-pick this commit on capa-testfiles

**Comment by mr-tz** [comment]
Good catch, done. Thank you!

**Pull request merged**